### ISSUE TITLE
Create L1 tests for Git

### ISSUE THREAD

**Issue by stephenmichaelf** [Microsoft/azure-devops-node-api#220]

**Comment by damccorm** [comment]
@stephenmichaelf As I've started working on this, I've started to question whether it might just make more sense to prioritize building out the samples (this applies to all of issues 219-222). Because we can't rely on what users' already have in their projects, to test effectively we have to create anything we want to test on. This in turn limits what we can test because there's not a way to programmatically create a lot of the things that we'd want to build test around (e.g. its not possible to create a commit via the GitApi, but that would be a central thing to test). With samples we can get more complete coverage and we already have a solid starting foundation.

I'm definitely not opposed to creating tests, I just think it might make sense to prioritize the samples. Maybe we can figure it out offline on Monday.
